src/app/marketplace/marketplace-product-details.component.ts

```diff
@@ -44,14 +65,43 @@ export class MarketplaceProductDetailsComponent implements OnInit, OnDestroy {
     return this.ipfsService.downloadAndSave(file.fileHash, file.fileName);
   }
 
+  checkDaysToDeliverLeft(date: Date): number {
+    if (!this.blockchainDataProductOrder) {
+      return 0;
+    }
+
+    return Math.ceil((this.blockchainDataProductOrder.deliveryDeadline.getTime() - date.getTime()) / DAY_IN_MILLISECONDS);
+  }
+
   ngOnInit(): void {
+    this.walletSubscription = this.walletService.getWallet().subscribe(wallet => {
+      this.wallet = wallet;
+
+      if (this.dataProductAddress) {
+        this.loadBlockchainDataProductOrder();
+      }
+    });
+
     this.activeRoute.params.subscribe(routeParams => {
+      this.dataProductAddress = routeParams.address;
       this.loadProduct(routeParams.address);
     });
+
+    this.clockSubscription = this.clockService.onEachSecond().subscribe(date => {
```

Comment:
I don't think that we should check days lest on each second. Please update to eachHour maybe.